### PR TITLE
server.rb - properly cork & uncork ssl clients

### PR DESCRIPTION
**Pull request by MSP-Greg**
### Description

`Server##cork_socket` & `Server##uncork_socket` work on TCPSockets, but when using an ssl client, a MiniSSL::Socket is passed to them.

Use 'to_io', which is available with all 'socket objects' used in Puma.

I haven't worked with wireshark for a while, and that's probably the only way to see a difference?

### Your checklist for this pull request
- [x] I have reviewed the [guidelines for contributing](../blob/master/CONTRIBUTING.md) to this repository.
- [x] I have added an entry to [History.md](../blob/master/History.md) if this PR fixes a bug or adds a feature. If it doesn't need an entry to HISTORY.md, I have added `[changelog skip]` or `[ci skip]` to the pull request title.
- [ ] I have added (or updated) appropriate tests if this PR fixes a bug or adds a feature.
- [x] My pull request is 100 lines added/removed or less so that it can be easily reviewed.
- [ ] If this PR doesn't need tests (docs change), I added `[ci skip]` to the title of the PR.
- [ ] If this closes any issues, I have added "Closes `#issue`" to the PR description or my commit messages.
- [ ] I have updated the documentation accordingly.
- [x] All new and existing tests passed, including Rubocop.

### CODE REVIEW

File: lib/puma/server.rb [hunk 1]
```diff
@@ -120,15 +120,15 @@ def current
...
+        RbConfig::CONFIG['host_os'].include?('linux') &&
           Socket.const_defined?(:IPPROTO_TCP) &&
           Socket.const_defined?(:TCP_CORK)
       end
 
       # :nodoc:
       # @version 5.0.0
       def closed_socket_supported?
-        RbConfig::CONFIG['host_os'] =~ /linux/ &&
+        RbConfig::CONFIG['host_os'].include?('linux') &&
```
Comment (nateberkopec) [inline lib/puma/server.rb:131]: What's the motivation for the change here?

Comment (MSP-Greg) [reply lib/puma/server.rb:131]: General code improvement.

1. If you need a boolean, choose a method that returns one.  I think that was the reason `String#match?` was added in Ruby 2.4.
2. Use string methods if a RegExp test isn't needed.  If and when macOS supports 'cork' (and hence `/linux|darwin/`), it can be a string method.

**Off topic:** A long time ago, my first 'significant' contribution to 'OS Ruby' was working on building & testing Windows/MinGW Ruby.  Since then, I've always kept an eye on the commits to ruby/ruby.  There are a lot of commits there that include small code changes like the change above.

I can revert...

Comment (MSP-Greg) [reply lib/puma/server.rb:131]: @nateberkopec

I looked at this a bit more, and `SOCKET::TCP_CORK` is not defined in Windows and macOS.  So, a better solution might be to add `HAS_TCP_CORK` to detect.rb and add a conditional to the method calls? So:
```ruby
cork_socket io
uncork_socket io
```

would change to:
```ruby
cork_socket(io) if HAS_TCP_SOCKET
uncork_socket(io) if HAS_TCP_SOCKET
```

Comment (MSP-Greg) [reply lib/puma/server.rb]: Ran `ruby -rsocket -e 'puts Socket.constants.sort'` in CI, and updated the code based on that.  The needed constants are not defined in Windows or macOS, so removed the linux detection...

File: History.md [hunk 2]
```diff
@@ -19,7 +21,6 @@
   * Fix phased restart errors related to nio4r gem when using the Puma control server ([#2516])
   * Add `#string` method to `Puma::NullIO` ([#2520])
   * Fix binding via Rack handler to IPv6 addresses ([#2521])
-  * Require rack/common_logger explicitly if :verbose is true ([#2547])
```
Comment (nateberkopec) [inline History.md:22]: whoops good catch
